api/channel.go

```diff
@@ -951,6 +952,33 @@ func getChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 
 }
 
+func getChannelByName(c *Context, w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	channelname := params["channel_name"]
+
+	cchan := Srv.Store.Channel().GetByName(c.TeamId, channelname)
+
+	if cresult := <-cchan; cresult.Err != nil {
+		c.Err = cresult.Err
+		return
+	} else {
+		data := &model.Channel{}
+		data = cresult.Data.(*model.Channel)
+
```

Comment:
Can you add a permission check here to make sure the user has read permission for this channel? Should be something like this:
```
if !HasPermissionToChannelContext(c, data.Id, model.PERMISSION_READ_CHANNEL) {
	return
}
```

Also can you update the unit test to test the permissions?